### PR TITLE
Fix API docs for unencoded hexaddress arguments

### LINKED ISSUE #2710: Be consistent with response codes in the docs
## Problem Definition
As a result of https://github.com/raiden-network/raiden/issues/2706 it is clear that some places in the docs are inconsistent with the actual responses provided by the API.

We need to go through the docs and make sure it is consistent all the way through according to the following rules:
- If there is an error in the URL return 404
- If there is malformed request data return 400
- If there are other errors in the values of the request then we just have 409 as a catch all.

## Tasks
- [ ] Go through the API doc and make sure it's consistent with the API code.
- [ ] If the API code is inconsistent somewhere fix the code

### PR DESCRIPTION
**Pull request by palango**
Fixes #2710

### CODE REVIEW

File: docs/rest_api.rst [hunk 1]
```diff
@@ -210,7 +210,7 @@ Querying Information About Channels and Tokens
       ]
 
    :statuscode 200: Successful query
-   :statuscode 400: The given token address is not valid eip55-encoded Ethereum address
+   :statuscode 404: The given token address is not valid eip55-encoded Ethereum address
```
Comment (czepluch) [inline docs/rest_api.rst]: Would say: "... is not **a** valid eip55..."

File: docs/rest_api.rst [hunk 2]
```diff
@@ -301,7 +302,7 @@ Querying Information About Channels and Tokens
 
    :statuscode 200: Successful query
    :statuscode 302: If the user accesses the channel link endpoint
-   :statuscode 404: If the token does not exist/the token address is invalid
+   :statuscode 404: If the token does not exist/the token address is not valid eip55-encoded
```
Comment (czepluch) [inline docs/rest_api.rst]: please consistent about space before and after slash. "exist/the" or "exist / the" as above

File: docs/rest_api.rst [hunk 3]
```diff
@@ -499,6 +501,7 @@ Connection Management
 
    :statuscode 204: For a successful connection creation.
    :statuscode 402: If any of the channel deposits fail due to insufficient ETH balance to pay for the gas of the on-chain transactions.
+   :statuscode 404: The given token and is not a valid eip55-encoded Ethereum address
```
Comment (czepluch) [inline docs/rest_api.rst]: I think you're missing a word here or you have a superfluous "and"

File: docs/rest_api.rst [hunk 4]
```diff
@@ -534,6 +537,7 @@ Connection Management
    The response is a list with the addresses of all closed channels.
 
    :statuscode 200: For successfully leaving a token network
+   :statuscode 404: The given token and is not a valid eip55-encoded Ethereum address
```
Comment (czepluch) [inline docs/rest_api.rst]: Same as above

File: docs/rest_api.rst [hunk 5]
```diff
@@ -701,8 +706,7 @@ The format of ``log_time`` is ISO8601 with milliseconds.
     ]
 
    :statuscode 200: For successful query
-   :statuscode 400: If the provided query string is malformed
-   :statuscode 404: If the token at the given token_address does not exist
+   :statuscode 404: The given token and is not a valid eip55-encoded Ethereum address or does not exist
```
Comment (czepluch) [inline docs/rest_api.rst]: Same as above

**Comment by czepluch** [comment]
Did you also check that this is indeed what the API returns in all the cases?

**Comment by palango** [comment]
Nice catches, should be fixed.
And yes, I tested the endpoints I changed.

**Pull request merged**